django_freeradius/api/views.py

```diff
@@ -161,3 +162,44 @@ def _get_status_type(self, request):
 
 
 accounting = AccountingView.as_view()
+
+
+class BatchView(generics.CreateAPIView):
+    queryset = RadiusBatch.objects.all()
+    serializer_class = RadiusBatchSerializer
+
+    def post(self, request, *args, **kwargs):
+        serializer = RadiusBatchSerializer(data=request.data)
+        if serializer.is_valid():
+            name = serializer.data.get('name')
+            expiration_date = serializer.data.get('expiration_data')
+            strategy = serializer.data.get('strategy')
+            if strategy == 'csv':
+                csvfile = request.FILES['csvfile']
+                batch = RadiusBatch(name=name,
+                                    strategy=strategy,
+                                    expiration_date=expiration_date,
+                                    csvfile=csvfile)
+                batch.csvfile_upload(csvfile)
+                response = RadiusBatchSerializer(batch)
+            elif strategy == 'prefix':
+                prefix = serializer.data.get('prefix')
+                batch = RadiusBatch(name=name,
+                                    strategy=strategy,
+                                    expiration_date=expiration_date,
+                                    prefix=prefix)
+                number_of_users = int(request.data['number_of_users'])
+                number_of_users_validated = self._validate_number_of_users(number_of_users)
+                batch.prefix_add(prefix, number_of_users_validated)
+                response = RadiusBatchSerializer(batch)
+            return Response(response.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    def _validate_number_of_users(self, number):
+        if number > 0:
+            return number
+        else:
+            raise ValidationError({'number_of_users': [_('Number of users should be greater than 0.')]})
```

Comment:
I think you can accomplish this much more easily because it's a feature already provided by django-rest-framework for integer fields, see the `min_value` argument of the serializer `IntegerField` here: http://www.django-rest-framework.org/api-guide/fields/#integerfield